control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -391,39 +404,220 @@ func (r Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkainterna
+		return fmt.Errorf("ScaledObject %q is not owned by %q", triggerAuth.Name, obj)
+	} else if !equality.Semantic.DeepDerivative(triggerAuth.Spec, expectedTriggerAuth.Spec) {
+		logging.FromContext(ctx).Debug(fmt.Sprintf("TriggerAuthentication changed, found: %#v expected: %#v", triggerAuth.Spec, expectedTriggerAuth.Spec))
+		triggerAuth.Spec = expectedTriggerAuth.Spec
+		if _, err = r.KedaClient.KedaV1alpha1().TriggerAuthentications(expectedTriggerAuth.Namespace).Update(ctx, triggerAuth, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+		return reconciler.NewEvent(corev1.EventTypeNormal, "TriggerAuthenticationUpdated", "TriggerAuthentication updated: \"%s/%s\"",
+			triggerAuth.Namespace, triggerAuth.Name)
+	} else {
+		logging.FromContext(ctx).Debugw("Reusing existing TriggerAuthentication", zap.Any("TriggerAuthentication", triggerAuth))
+	}
+
+	return nil
+}
+
+func (r *Reconciler) reconcileSecret(ctx context.Context, expectedSecret *corev1.Secret, obj metav1.Object) error {
+	secret, err := r.KubeClient.CoreV1().Secrets(expectedSecret.Namespace).Get(ctx, expectedSecret.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		secret, err = r.KubeClient.CoreV1().Secrets(expectedSecret.Namespace).Create(ctx, expectedSecret, metav1.CreateOptions{})
+		if err != nil {
+			logging.FromContext(ctx).Errorw("Unable to create Secret", zap.Error(err))
+			return reconciler.NewEvent(corev1.EventTypeWarning, "SecretDeploymentFailed", "Secret deployment failed to: \"%s/%s\", %w",
+				expectedSecret.Namespace, expectedSecret.Name, err)
+		}
+		return reconciler.NewEvent(corev1.EventTypeNormal, "SecretCreated", "Secret created: \"%s/%s\"", secret.Namespace, secret.Name)
+	} else if err != nil {
+		logging.FromContext(ctx).Errorw("Unable to get an existing Secret", zap.Error(err))
+		return err
+	} else if !metav1.IsControlledBy(secret, obj) {
+		return fmt.Errorf("secret %q is not owned by %q", secret.Name, obj)
+	} else {
+		// StringData is not populated on read so for now always update the secret
+		logging.FromContext(ctx).Debug("Updating secret")
+		secret.StringData = expectedSecret.StringData
+		if _, err = r.KubeClient.CoreV1().Secrets(expectedSecret.Namespace).Update(ctx, secret, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+		return reconciler.NewEvent(corev1.EventTypeNormal, "SecretUpdated", "Secret updated: \"%s/%s\"", secret.Namespace, secret.Name)
+	}
+}
+
+func (r *Reconciler) isKEDAEnabled(ctx context.Context, namespace string) bool {
+	// TODO: code below failing unit tests with err: "panic: interface conversion: testing.ActionImpl is not testing.GetAction: missing method GetName"
+	/*if err := discovery.ServerSupportsVersion(r.KubeClient.Discovery(), keda.KedaSchemeGroupVersion); err == nil {
+		return true
+	}*/
+
+	if _, err := r.KedaClient.KedaV1alpha1().ScaledObjects(namespace).List(ctx, metav1.ListOptions{}); err == nil && r.KafkaFeatureFlags.IsControllerAutoscalerEnabled() {
```

Comment:
To save an API call when the flag is not enabled:
```suggestion
	if r.KafkaFeatureFlags.IsControllerAutoscalerEnabled() {
	        _, err := r.KedaClient.KedaV1alpha1().ScaledObjects(namespace).List(ctx, metav1.ListOptions{});
		return err == nil
	}
```